python/cog/server/telemetry.py

```diff
@@ -0,0 +1,44 @@
+from typing import TypedDict, Union
+from contextlib import contextmanager
+from contextvars import ContextVar
+
+# See: https://www.w3.org/TR/trace-context/
+TraceContext = TypedDict(
+    "TraceContext", {"traceparent": Union[str, None], "tracestate": Union[str, None]}
+)
+
+TRACE_CONTEXT: ContextVar[Union[TraceContext, None]] = ContextVar(
+    "trace_context", default=None
+)
+
+
+def make_trace_context(
+    traceparent: Union[str, None] = None, tracestate: Union[str, None] = None
+) -> TraceContext:
+    """
+    Creates a trace context dictionary from the given traceparent and tracestate
+    headers. This is used to pass the trace context between services.
+    """
+    return {"traceparent": traceparent, "tracestate": tracestate}
```

Comment:
I'm not sure you want to set these in the object if they're none. As written you'll be setting headers later to the literal string `"None"`.

Reply:
Hmm, good catch, I was getting frustrated with the typechecker. But I see `TypedDict` takes a `total` keyword that does what I need.